pages/animated_card_page.py

```diff
@@ -0,0 +1,132 @@
+        elif field_type == FieldType.SECURITY_CODE.name:
+            if card_type == "AMEX":
+                animated_card_data = self._action.get_text(self._locators.cvv_on_front_side_animated_card)
+            else:
+                animated_card_data = self._action.get_text(self._locators.cvv_on_back_side_animated_card)
+        return animated_card_data
+
+    def validate_data_on_animated_card(self, expected_data, field_type, card_type):
+        actual_data_on_animated_card = self.get_data_from_animated_card(field_type, card_type)
+        assert expected_data in actual_data_on_animated_card, f'Data on animated card is not correct, should be: ' \
+                                                              f'"{expected_data}" but is: "{actual_data_on_animated_card}"'
+
+    def validate_all_data_on_animated_card(self, card_number, exp_date, cvv, card_type, is_field_in_iframe):
+        if is_field_in_iframe:
+            self._action.switch_to_iframe(FieldType.ANIMATED_CARD.value)
+        self.validate_data_on_animated_card(card_number, FieldType.CARD_NUMBER.name, card_type)
+        self.validate_data_on_animated_card(exp_date, FieldType.EXPIRATION_DATE.name, card_type)
+        if cvv is not None:
+            self.validate_data_on_animated_card(cvv, FieldType.SECURITY_CODE.name, card_type)
+
+    def validate_if_animated_card_is_flipped(self, card_type, is_field_in_iframe):
+        if is_field_in_iframe:
+            self._action.switch_to_iframe(FieldType.ANIMATED_CARD.value)
+        animated_card_side = self._action.get_element_attribute(self._locators.animated_card, "class")
+        if card_type == "AMEX":
+            assert "flip-card" not in animated_card_side, f'Animated card is flipped for AMEX but should not be'
+        else:
+            assert "flip-card" in animated_card_side, f'Animated card is not flipped but should be'
+
+    def change_field_focus(self):
+        self._action.click(self._locators.card_number_input_field)
+
+    def validate_if_no_iframe_field_is_highlighted(self, field_type):
+        is_highlighted = self.is_field_highlighted(field_type)
+        assert is_highlighted is True, f'{FieldType[field_type].name} field is not highlighted but should be'
+
+    def is_field_highlighted(self, field_type):
+        is_highlighted = False
+        class_name = ""
+        if field_type == FieldType.CARD_NUMBER.name:
+            class_name = self._action.get_element_attribute(self._locators.card_number_input_field, "class")
+        elif field_type == FieldType.EXPIRATION_DATE.name:
+            class_name = self._action.get_element_attribute(self._locators.expiration_date_input_field, "class")
+        elif field_type == FieldType.SECURITY_CODE.name:
+            class_name = self._action.get_element_attribute(self._locators.security_code_input_field, "class")
+        if class_name.__contains__("error"):
```

Comment:
it's more pythonic to say `if "error" in class_name:`

Reply:
FYI theres a few places that call __contains__ which can be changed to use the `in`

Reply:
ok, I will change them all